planning/behavior_path_planner/src/path_utilities.cpp

```diff
@@ -89,80 +90,40 @@ double calcPathArcLength(const PathWithLaneId & path, size_t start, size_t end)
  */
 PathWithLaneId resamplePathWithSpline(const PathWithLaneId & path, double interval)
 {
-  const auto base_points = calcPathArcLengthArray(path);
-  const auto sampling_points = tier4_autoware_utils::arange(0.0, base_points.back(), interval);
-
-  if (base_points.empty() || sampling_points.empty()) {
+  if (path.points.size() < 2) {
     return path;
   }
 
-  std::vector<double> base_x, base_y, base_z;
-  for (const auto & p : path.points) {
-    const auto & pos = p.point.pose.position;
-    base_x.push_back(pos.x);
-    base_y.push_back(pos.y);
-    base_z.push_back(pos.z);
+  double s = 0.0;
+  std::vector<double> s_in{0.0};
+  for (size_t i = 0; i < path.points.size() - 1; ++i) {
+    s += tier4_autoware_utils::calcDistance2d(
+      path.points.at(i).point.pose, path.points.at(i + 1).point.pose);
+    s_in.push_back(s);
   }
 
-  const auto resampled_x = interpolation::slerp(base_points, base_x, sampling_points);
-  const auto resampled_y = interpolation::slerp(base_points, base_y, sampling_points);
-  const auto resampled_z = interpolation::slerp(base_points, base_z, sampling_points);
+  std::vector<double> s_out = s_in;
 
-  PathWithLaneId resampled_path{};
-  resampled_path.header = path.header;
-  resampled_path.drivable_area = path.drivable_area;
-
-  // For Point X, Y, Z
-  for (size_t i = 0; i < sampling_points.size(); ++i) {
-    PathPointWithLaneId p{};
-    p.point.pose.position =
-      tier4_autoware_utils::createPoint(resampled_x.at(i), resampled_y.at(i), resampled_z.at(i));
-    resampled_path.points.push_back(p);
-  }
+  constexpr double epsilon = 0.01;
+  const auto has_almost_same_value = [&](const auto & vec, const auto x) {
```

Comment:
@purewater0901 
[Q] how was almost_same_value(point) handled in behavior path planner previously?

Reply:
They previously do not sample input points, so it does not handle duplicated points